packages/tinybased/src/lib/search/generateSearched.tsx

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import { SearchParams, SearchResult } from '@lyrasearch/lyra';
+
+import { useCallback, useEffect, useState } from 'react';
+import { useValue } from 'tinybase/cjs/ui-react';
+import { ObjectToCellStringType, TableBuilder } from '../TableBuilder';
+import { TinyBased } from '../tinybased';
+import { OnlyStringKeys, TinyBaseSchema } from '../types';
+import { Searched } from './searched';
+
+const SEARCH_LAST_UPDATED_AT = (table: string | symbol | number) =>
+  `${String(table)}-last-updated-at`;
+
+/**
+ * Generates a type-safe Searched instance and its corresponding `useSearch` hook for a fully
+ * connected and reactive Full Text Search of an underlying `tinybased` dataset.
+ */
+export const generateSearched = async <
```

Comment:
I feel like we could have more compelling name for this since its the thing that is providing high level glue between two really powerful components that we've developed. In its current state it sounds like a generic factory function for creating a Search Index but its doing way more cool stuff than that under the hood

Maybe something cool like `searchifyTinybased`? I'm open to suggestions

Reply:
I've elected on `tinybaseSearcher` as the output is a searcher instance (and related useSearch hook) for that tinybased instance.

`Searchify` sound to me like we're adding search as a plugin to `tinybased` and should expect more functionality from `tinybased` instead of the added functionality coming from the return of this function.

Reply:
After mulling it over more, maybe `connectTinybasedSearcher` to hint as to the action it performs (connecting a Searcher to our Tinybased).